web/src/portico/landing-page.ts

```diff
@@ -118,13 +135,15 @@ const apps_events = function () {
     update_page();
 };
 
-const events = function () {
+const events = function (): void {
     // get the location url like `zulip.com/features/`, cut off the trailing
     // `/` and then split by `/` to get ["zulip.com", "features"], then
     // pop the last element to get the current section (eg. `features`).
     const location = window.location.pathname.replace(/\/$/, "").split(/\//).pop();
 
-    $(`[data-on-page='${CSS.escape(location)}']`).addClass("active");
+    if (location) {
+        $(`[data-on-page='${CSS.escape(location)}']`).addClass("active");
+    }
```

Comment:
Would a type assertion that we expect this to be defined be more appropriate here?

Reply:
@timabbott Technically everywhere in codebase where such behaviour exists, we should just use type assertion. Since the code works there's no point of adding strict type checking unless we're planning to induce some other behaviour when strict type checking fails. For example returning/throwing errors, triggering an event or halting the function execution etc.

Reply:
You probably shouldn’t trust that previous implementers have thought much about `undefined` while working in a language that usually lets them get away with incorrectly ignoring it. That’s a large part of why we want to convert to TypeScript.

In this particular case, `location` can’t be `undefined` because `split()` never returns an empty array, so a non-null assertion would be appropriate. Alternatively, we could start using the [`nullthrows` library](https://www.npmjs.com/package/nullthrows) to abbreviate the assertion pattern.

Reply:
@andersk Ahh you're correct, `split()` at the very least returns one item like `[""]` so `.pop()` will pop `""`. Didn't know this, thanks for letting me know. I guess the `@types/node` ts bindings are not implemented to infer this behaviour.

Reply:
@andersk Used type assertion.